src/NotificationTargetCommonITILObject.php

```diff
@@ -1423,7 +1423,7 @@ public function getDataForObject(CommonDBTM $item, array $options, $simple = fal
                     ]
                 ],
                 'WHERE'     => [
-                    $item->getAssociatedDocumentsCriteria(),
+                    $item->getAssociatedDocumentsCriteria(Session::isCron()),
```

Comment:
We should never rely on session in notification process. Indeed, restrictions should depend on the notification target, not on the user that triggers the notification.
For instance, if a followup is added by a user that has no rights to see solutions, the documents related to solutions will not be available in notifications sent to a super admin.

As we already bypass rights when attaching documents to the sent email, we could use the same logic here.
```suggestion
                    $item->getAssociatedDocumentsCriteria(true),
```

It may expose documents that should not, but this is already the case in `NotificationEventMailing`. The only way to fix this is to do a huge refactoring of the notification building to be able to build visibility restrictions based on a user ID, instead of building them on current session or on bypassing all restrictions, but it will be hard to implement it on a 10.0.x version.